src/spdx/parser/tagvalue/parser/tagvalue.py

```diff
@@ -0,0 +1,689 @@
+
+    @grammar_rule("pkg_checksum : PKG_CHECKSUM CHECKSUM")
+    def p_pkg_checksum(self, p):
+        self.check_that_current_element_matches_class_for_value(Package, p.lineno(1))
+        checksum = parse_checksum(self.current_element["logger"], p[2], p.lineno(1))
+        self.current_element.setdefault("checksums", []).append(checksum)
+
+    @grammar_rule("verification_code : PKG_VERF_CODE LINE")
+    def p_pkg_verification_code(self, p):
+        self.check_that_current_element_matches_class_for_value(Package, p.lineno(1))
+        if str(p.slice[0]) in self.current_element:
+            self.current_element["logger"].append(f"Multiple values for {p[1]} found. Line: {p.lineno(1)}")
+            return
+        verif_code_regex = re.compile(r"([0-9a-f]{40})\s*(\(excludes:\s*(.+)\))?", re.UNICODE)
+        verif_code_code_grp = 1
+        verif_code_exc_files_grp = 3
+        match = verif_code_regex.match(p[2])
+        if not match:
+            self.current_element["logger"].append(
+                f"Error while parsing {p[1]}: Value did not match expected format. Line: {p.lineno(1)}")
+            return
+        value = match.group(verif_code_code_grp)
+        excluded_files = None
+        if match.group(verif_code_exc_files_grp):
+            excluded_files = match.group(verif_code_exc_files_grp).split(",")
+        self.current_element[str(p.slice[0])] = PackageVerificationCode(value, excluded_files)
+
+    @grammar_rule("files_analyzed : PKG_FILES_ANALYZED LINE")
+    def p_pkg_files_analyzed(self, p):
+        self.check_that_current_element_matches_class_for_value(Package, p.lineno(1))
+        if str(p.slice[0]) in self.current_element:
+            self.current_element["logger"].append(f"Multiple values for {p[1]} found. Line: {p.lineno(1)}")
+            return
+        self.current_element[str(p.slice[0])] = p[2] in ['true', 'True', True]
+
+    @grammar_rule("pkg_file_name : PKG_FILE_NAME LINE")
+    def p_pkg_file_name(self, p):
+        self.check_that_current_element_matches_class_for_value(Package, p.lineno(1))
+        set_value(p, self.current_element, argument_name="file_name")
+
+    @grammar_rule("package_version : PKG_VERSION LINE")
+    def p_package_version(self, p):
+        self.check_that_current_element_matches_class_for_value(Package, p.lineno(1))
+        set_value(p, self.current_element, argument_name="version")
+
+    @grammar_rule("primary_package_purpose : PRIMARY_PACKAGE_PURPOSE primary_package_purpose_value")
+    def p_primary_package_purpose(self, p):
+        self.check_that_current_element_matches_class_for_value(Package, p.lineno(1))
+        set_value(p, self.current_element, method_to_apply=lambda x: PackagePurpose[x.replace("-", "_")])
+
+    @grammar_rule("primary_package_purpose_value : APPLICATION\n | FRAMEWORK\n | LIBRARY\n | CONTAINER\n "
+                  "| OPERATING_SYSTEM \n | DEVICE \n| FIRMWARE\n | SOURCE\n | ARCHIVE\n | FILE\n | INSTALL\n | OTHER")
```

Comment:
The html spec only allows `OPERATING-SYSTEM` whereas the json spec allows both values. I am in favor of staying close to the html spec (which should be single source of truth for tag value) and not allowing both cases. This could open a discussion on whether we should always allow `_` and `-`...
Probalby the confusion here is that the token for the value `OPERATING-SYSTEM` is `OPERATING_SYSTEM`, I will change this.